File: src/main/webapp/js/pago.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */
/*
 * pago.js — Gestión de Pagos
 */

function cargarPagos() {
    fetch("PagoServlet")
        .then(res => res.json())
        .then(pintarTabla);
}

function pintarTabla(lista) {

    let tabla = `
    <table class="table table-bordered table-striped table-hover">
        <tr class="table-dark">
            <th>#</th>
            <th>Alumno</th>
            <th>Servicio</th>
            <th>Monto</th>
            <th>Fecha Pago</th>
            <th>Método Pago</th>
            <th>Estado Pago</th>
        </tr>`;

    lista.forEach(pago => {
        tabla += `
        <tr>
            <td>${pago.idPago}</td>
            <td>${pago.alumno}</td>
            <td>${pago.servicio}</td>
            <td>$${pago.monto}</td>
            <td>${pago.fechaPago}</td>
            <td>${pago.metodoPago}</td>
            <td>${pago.estadoPago}</td>
        </tr>`;
    });

    tabla += "</table>";

    $("#listaPagos").html(tabla);
}

cargarPagos();

File: src/main/java/com/ues/edu/modelo/Pago.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.ues.edu.modelo;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import lombok.Data;
import lombok.NoArgsConstructor;
/**
 *
 * @author kikej
 */

@Data
@NoArgsConstructor
@Entity
@Table(name = "pagos")
public class Pago {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_pago")
    private Integer idPago;

    @NotNull(message = "La inscripción no puede ser nula")
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_inscripcion", nullable = false)
    private Inscripcion inscripcion;

    @NotNull(message = "El monto no puede ser nulo")
    @DecimalMin(value = "0.0", inclusive = false, message = "El monto debe ser mayor a cero")
    @Column(name = "monto", nullable = false, precision = 10, scale = 2)
    private BigDecimal monto;

    @Column(name = "fecha_pago")
    private LocalDate fechaPago;

    @NotBlank(message = "El método de pago no puede estar vacío")
    @Column(name = "metodo_pago", length = 50)
    private String metodoPago;

    @Column(name = "estado_pago", nullable = false)
    private boolean estadoPago = false;

    @OneToOne(mappedBy = "pago", fetch = FetchType.LAZY)
    private Comprobante comprobante;

    @OneToMany(mappedBy = "pago", fetch = FetchType.LAZY)
    private List<Asistencia> asistencias;
}
